tools/langs/manuscript.py

```diff
@@ -156,8 +169,15 @@
 GetId "(element) {
         return element._id;
 }"
-SetId "(element, value) {
-        element._id = value;
+SetId "(element, newId) {
+    UnregisterId(element._id);
+    element._id = value;
+    Self.MEIFlattened[value] = element;
+}"
+UnregisterId "(id) {
+    olddict = Self._property:MEIFlattened;
+    newdict = removeKeyFromDictionary(olddict, oldid);
```

Comment:
And here isn't `oldid` supposed to be the value passed in, `id`?

Reply:
That's right. I made sure I unloaded/reloaded the proper files in Sibelius and ran the tests successfully.